diesel_codegen/src/update.rs

```diff
@@ -138,14 +138,15 @@ fn save_changes_impl(
         let pk_field = pk.field_name.unwrap();
         quote_item!(cx,
             impl<'a> $struct_name {
-                $_pub fn save_changes<T>(&self, connection: &::diesel::Connection)
+                $_pub fn save_changes<T, Conn>(&self, connection: &Conn)
```

Comment:
Just need to pull this up to a `SaveChangesDsl<Conn: Connection>` and impl that instead.